fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/PermissiveAvailability.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.metadata.DataSourceMetadataService;
+import com.yahoo.bard.webservice.table.Column;
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+
+import org.joda.time.Interval;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * An availability which allows multiple disconnected intervals(union) on its contents.
+ */
+public class PermissiveAvailability implements Availability {
+
+    private final TableName tableName;
+    private final Set<Column> columns;
+    private final DataSourceMetadataService metadataService;
+    private final Set<TableName> dataSourceNames;
+
+    /**
+     * Constructor.
+     *
+     * @param tableName The name of the data source
+     * @param columns A set of columns associated with the data source
+     * @param metadataService A service containing the data source segment data
+     */
+    public PermissiveAvailability(
+            TableName tableName,
+            @NotNull Set<Column> columns,
+            @NotNull DataSourceMetadataService metadataService
+    ) {
+        this.tableName = tableName;
+        this.columns = columns;
+        this.metadataService = metadataService;
+        dataSourceNames = Collections.singleton(tableName);
+    }
+
+    @Override
+    public Set<TableName> getDataSourceNames() {
+        return dataSourceNames;
+    }
+
+    @Override
+    public Map<Column, Set<Interval>> getAllAvailableIntervals() {
+        Map<String, Set<Interval>> allAvailableIntervals = metadataService.getAvailableIntervalsByTable(tableName);
+
+        return columns.stream()
+                .collect(
+                        Collectors.toMap(
+                                Function.identity(),
+                                column -> allAvailableIntervals.getOrDefault(column.getName(), Collections.emptySet())
+                        )
+                );
+    }
+
+    @Override
+    public SimplifiedIntervalList getAvailableIntervals(DataSourceConstraint constraints) {
+        Set<String> allColumnNames = constraints.getAllColumnNames();
+
+        if (allColumnNames.isEmpty()) {
+            throw new IllegalArgumentException("DataSourceConstraint needs to have at least 1 constraint column");
```

Comment:
Follow the behavior if things are empty in `ConcreteAvailability`, we do not want different behavior for each Availability.